tests/time_tests/common/statistics_writer.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <cstdio>
+
+
+class StatisticsWriter {
+private:
+    std::ofstream statistics_file;
+
+    StatisticsWriter(const std::string &statistics_path) {
+        statistics_file.open(statistics_path);
+        if (!statistics_file.good()) {
+            std::stringstream err;
+            err << "Statistic file \"" << statistics_path << "\" can't be used for writing";
+            throw std::runtime_error(err.str());
+        }
+    }
+
+    StatisticsWriter(const StatisticsWriter&) = delete;
+    StatisticsWriter& operator=(const StatisticsWriter&) = delete;
+public:
+    static StatisticsWriter& Instance(){
+        static StatisticsWriter writer(FLAGS_s);
```

Comment:
This will reopen log at every Instance() call. I expect the log opened once.

Reply:
No, the object will be created once, so the file will be opened once too

Reply:
Replaced passing object to a ctor with `setFile` function. Additional checks have been added too